src/main/kotlin/com/exactpro/th2/lwdataprovider/db/CradleMessageExtractor.kt

```diff
@@ -338,11 +339,21 @@ internal class GroupBatchCheckIterator(
 ) : Iterator<StoredGroupedMessageBatch> by original {
     override fun next(): StoredGroupedMessageBatch = original.next().also { batch ->
         if (batch.messages.size > 1) {
-            batch.messages.reduce { first, second ->
-                check(first.timestamp <= second.timestamp) {
-                    "Unordered message received for: ${batch.toShortInfo()} batch, between ${first.toShortInfo()} and ${second.toShortInfo()} messages"
+            val map = hashMapOf<Stream, StreamMetadata>()
+            batch.messages.forEach { message ->
+                val stream = Stream(message.sessionAlias, message.direction)
+                val previous = map[stream] ?: MIN_STREAM_METADATA
```

Comment:
We can remove setting the value at the end of the check
```suggestion
                val previous = map.put(stream, StreamMetadata(message.timestamp, message.sequence))
                                         ?: MIN_STREAM_METADATA
```